cassandra-executor/src/main/java/io/mesosphere/mesos/frameworks/cassandra/jmx/Nodetool.java

```diff
@@ -0,0 +1,118 @@
+/**
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.mesosphere.mesos.frameworks.cassandra.jmx;
+
+import java.net.UnknownHostException;
+import java.util.Map;
+
+public class Nodetool {
+    private final JmxConnect jmxConnect;
+
+    public Nodetool(JmxConnect jmxConnect) {
+        this.jmxConnect = jmxConnect;
+    }
+
+    public boolean isNativeTransportRunning() {
+        return jmxConnect.getStorageServiceProxy().isNativeTransportRunning();
+    }
+
+    public boolean isRPCServerRunning() {
+        return jmxConnect.getStorageServiceProxy().isRPCServerRunning();
+    }
+
+    /**
+     * Whether the node has joined the ring.
+     */
+    public boolean isJoined() {
+        return jmxConnect.getStorageServiceProxy().isJoined();
+    }
+
+    /**
+     * to determine if gossip is disabled
+     */
+    public boolean isGossipInitialized() {
+        return jmxConnect.getStorageServiceProxy().isInitialized();
+    }
+
+    /**
+     * whether gossip is running or not
+     */
+    public boolean isGossipRunning() {
+        return jmxConnect.getStorageServiceProxy().isGossipRunning();
+    }
+
+    /**
+     * Get the operational mode.
+     * Valid operation modes are:
+     * <ul>
+     *     <li>STARTING</li>
+     *     <li>NORMAL</li>
+     *     <li>JOINING</li>
+     *     <li>LEAVING</li>
+     *     <li>DECOMMISSIONED</li>
+     *     <li>MOVING</li>
+     *     <li>DRAINING</li>
+     *     <li>DRAINED</li>
+     * </ul>
+     */
+    public String getOperationMode() {
```

Comment:
looks like - but (although unlikely) an enum would not cover a new operation mode.